service/src/test/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsServiceTest.java

```diff
@@ -0,0 +1,254 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentSocialWorkOther;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.DocumentUploadHelper;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.time.LocalDate.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.caseData;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.populatedCaseDetails;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {
+    JacksonAutoConfiguration.class,
+    UploadDocumentsService.class,
+    FixedTimeConfiguration.class
+})
+class UploadDocumentsServiceTest {
+
+    private static final String USER = "HMCTS";
+    private CaseData givenCaseData;
+
+    @Autowired
+    private Time time;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private UploadDocumentsService uploadDocumentsService;
+
+    @MockBean
+    private DocumentUploadHelper documentUploadHelper;
+
+    @BeforeEach
+    void setup() {
+        givenCaseData = prepareCaseData();
+        when(documentUploadHelper.getUploadedDocumentUserDetails()).thenReturn(USER);
+    }
+
+    @Test
+    void shouldSetUploadedByAndDateTimeOnCaseDocuments() {
+        CaseData caseData = caseData();
+
+        Map<String, Object> map = uploadDocumentsService.updateCaseDocuments(caseData, caseData);
```

Comment:
should we have separate instances of caseData?

Reply:
Not necessary I think. Point of the test is to see the result with uploadedBy and DateTime.

Reply:
In that case we could pass caseDataBefore as emptyList in this test and test scenario where user is uploading initial documents.